spec/client_spec.rb

```diff
@@ -36,13 +37,15 @@ def send_stream_content(res, content, keep_open:)
     res.content_type = "text/event-stream"
     res.status = 200
     res.chunked = true
-    rd, wr = IO.pipe
-    wr.write(content)
-    res.body = rd
-    if !keep_open
-      wr.close
+    if keep_open
+      rd, wr = IO.pipe
+      res.body = rd
+      wr.write(content)
+      wr
+    else
+      res.body = proc { |out| out.write(content) }
+      nil
```

Comment:
For unknown reasons, in JRuby sometimes using `IO.pipe` with an HTTP response and then closing the write pipe will cause the HTTP client to fail without even reading the beginning of the response. I changed it so that if we're going to close the response rather than keep it open, we don't use a pipe.